feat(cce): add datasource huaweicloud_cce_cluster_certificate (#3905)
huaweicloud/terraform-provider-huaweicloud: docs/data-sources/cce_cluster_certificate.md

---
subcategory: "Cloud Container Engine (CCE)"
---

# huaweicloud_cce_cluster_certificate

Provides details about the cluster certificate within HuaweiCloud.

## Example Usage

```hcl
variable "cluster_id" {}

data "huaweicloud_cce_cluster_certificate" "test" {
  cluster_id = var.cluster_id
  duration   = 30
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to obtain the CCE cluster certificate. If omitted, the
  provider-level region will be used.

> * `cluster_id` - (Required, String) Specifies the cluster id which the cluster certificate in.

Reviewer: ```suggestion
* `cluster_id` - (Required, String) Specifies the cluster ID which the cluster certificate in.

```

---
subcategory: "Cloud Container Engine (CCE)"
---

# huaweicloud_cce_cluster_certificate

> Provides details about the cluster certificate within HuaweiCloud.

Reviewer: ```suggestion
Use this data source to get the certificate of a CCE cluster within HuaweiCloud.

```

## Example Usage

```hcl
variable "cluster_id" {}

data "huaweicloud_cce_cluster_certificate" "test" {
  cluster_id = var.cluster_id
  duration   = 30
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to obtain the CCE cluster certificate. If omitted, the
  provider-level region will be used.

* `cluster_id` - (Required, String) Specifies the cluster id which the cluster certificate in.

> * `duration` - (Required, Int) Specifies the duration of the cluster certificate. The valid value in [1, 1827]. If the
>   input value is -1, it will use the maximum 1827 as `duration` value.

Reviewer: add a description of the duration unit
```suggestion
* `duration` - (Required, Int) Specifies the duration of the cluster certificate. The valid value range is [1, 1827]. If the
  input value is -1, it will use the maximum 1827 as `duration` value.

```

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

> * `id` - The certificate ID in UUID format.

Reviewer: it's not the certificate ID
```suggestion
* `id` - The data source ID in UUID format.

```
